spec/classes/puppet_config_spec.rb

```diff
@@ -138,16 +138,46 @@
         end
       end
 
-      describe "when use_srv_records => true" do
-        let :pre_condition do
-          "class { 'puppet': use_srv_records => true }"
+      describe "when use_srv_records => true, and domain fact" do
+        context 'is defined' do
+          let :pre_condition do
+            "class { 'puppet': use_srv_records => true }"
+          end
+
+          it 'should contain puppet.conf [main] with SRV settings' do
+            should contain_puppet__config__main("use_srv_records").with({'value' => "true"})
+            should contain_puppet__config__main("srv_domain").with({'value' => "example.org"})
+            should contain_puppet__config__main("pluginsource").with({'value' => "puppet:///plugins"})
+            should contain_puppet__config__main("pluginfactsource").with({'value' => "puppet:///pluginfacts"})
+          end
+        end
+
+        context 'is unset' do
+          let(:facts) { facts.merge({domain: nil}) }
+          let :pre_condition do
+            'class { ::puppet:
+               use_srv_records => true
+            }'
+          end
+
+          it 'should fail with a helpful message' do
+            should_not compile
+          end
         end
 
-        it 'should contain puppet.conf [main] with SRV settings' do
-          should contain_puppet__config__main("use_srv_records").with({'value' => "true"})
-          should contain_puppet__config__main("srv_domain").with({'value' => "example.org"})
-          should contain_puppet__config__main("pluginsource").with({'value' => "puppet:///plugins"})
-          should contain_puppet__config__main("pluginfactsource").with({'value' => "puppet:///pluginfacts"})
+        context 'is overriden via param' do
+          let :pre_condition do
+            'class { "::puppet":
+               use_srv_records => true,
+               srv_domain      => "special_domain.com"
+            }'
+          end
+
+          it 'should configure srv domain' do
+            should compile.with_all_deps
```

Comment:
You may need to leave this out, since validation of the catalog fails when testing Windows (since it runs on Linux), causing test failures.